Clean branch of projects v2

zoek1 opened the pull request:
<!-- 
Thank you for your pull request! Please review the requirements below, read through the contributor's guide, 
and ensure your pull request has fulfilled all requirements outlined by the Gitcoin Core team.
Have you read the contributors guide?: https://docs.gitcoin.co/mk_contributors/ 
-->

##### Description
![image](https://user-images.githubusercontent.com/660973/93838497-755f0980-fc4f-11ea-9ba8-d08961f808d1.png)
![image](https://user-images.githubusercontent.com/660973/93838583-c242e000-fc4f-11ea-8748-c83021a20dc6.png)
https://www.loom.com/share/c7f78645ca2a4219b36aa3475564278d
<!-- Describe your changes here. -->

##### Refers/Fixes
https://github.com/gitcoinco/web/issues/7340
<!-- If this PR is related to a Github issue, please add a link here. -->

##### Testing

<!-- All PRs should be accompanied by tests! If you haven't added tests, please explain here. -->

zoek1 commented:
@thelostone-mc this replace https://github.com/gitcoinco/web/pull/7493, only remove the dependency of `convert2grant` also address Octavio's comments :+1:

thelostone-mc commented on app/dashboard/templates/dashboard/hackathon/project_new.html:
```diff
@@ -70,6 +70,40 @@ <h2 class="h5 text-center font-weight-bold mb-4">Let's Get Started!</h2>
                 <label  class="font-weight-semibold" for="work_url">Project Github Repository or Link to Pull Request</label>
                 <input type="url" class="form-control" name="work_url" id="work_url" value="{{project_selected.work_url}}" placeholder="https://github.com/gitcoinco/web" required>
+              </div>
+                            <div class="form-group mt-4">
```
fix up indenation

danlipert replied:
yah lets plz fix this @zoek1

thelostone-mc commented on app/dashboard/views.py:
```diff
@@ -4485,12 +4499,21 @@ def hackathon_save_project(request):
     project = HackathonProject.objects.filter(pk=project_id).nocache().first()
     if not project:
-        raise Http404("The project doesnt exists.")
+        raise None
```
^ why are we changing this ?

chibie replied:
@thelostone-mc cause `project` isn't a view function. it's used in `get_project` view @ line 4502

thelostone-mc reviewed (changes requested):
travis failing

chibie commented:
@zoek1 after editing a project, page reload should be automatic instead of doing so manually to view effected changes.

zoek1 commented:
@chibie did you tested with the latest changes?

chibie commented:
@zoek1 yea, from about 16 hours ago.

chibie commented on app/assets/v2/js/hackathon-projects.js:
```diff
@@ -68,6 +72,15 @@ const projectModal = (bountyId, projectId, callback) => {
     });
   });
 
+  $('#videodemo-url').keyup(function(event) {
```
@zoek1 it seems this block of code is unreached cause the video url isn't saved when i submit.

chibie replied:
i get `Error in render: "TypeError: Cannot read property 'url' of undefined` in the console when i submit.

zoek1 replied:
There's two things here: the first is keyup listen on changes on the field with the id #videodemo-url, this doesn't requires an existing value, it will evaluate while the user write something on such field. 
The second Is a good catch, i fixed that with the latest commit also that prevent refresh the data in the page. Thanks for review It!

chibie replied:
@zoek1 

it second issue has been resolved. 👍 💯  

for the first issue about video not saving in db, kindly observe the recording below,

https://www.loom.com/share/406b9513467c4fb7967362ba152b9c4b

if `keyup` doesn't work as expected, don't you think you should `getVideoMetadata` on submit instead?

ps: i tested on google chrome browser.

thelostone-mc replied:
^ @PixelantDesign / @zoek1 will have to wait until this gets resolved

octavioamu commented on app/dashboard/templates/addinterest.html:
```diff
@@ -124,6 +124,18 @@ <h5 class="text-center font-title">{% trans "Submit a Plan" %}</h5>
                 <input type="url" class="form-control" name="work_url" id="work_url" placeholder="https://github.com/gitcoinco/web" required>
               </div>
 
+              <div class="form-group mt-4">
+                <label class="font-weight-semibold" for="videodemo_url">Select video provider</label>
+                <select name="videodemo-provider">
+                  <option value=""><i class="fa fa-youtube"></i> Youtube</option>
+                  <option value=""><i class="fa fa-vimeo">Vimeo</option>
+                  <option value=""><i class="fa fa-loom">Loom</option>
+                  <option value=""><i class="fa fa-video">HTML5 video</option>
+                </select>>
+                <label class="font-weight-semibold" for="videodemo-url">Link to Video Demo</label>
+                <input type="url" class="form-control" name="work_url" id="work_url" placeholder="https://github.com/gitcoinco/web" required>
+              </div>
+
```
if someone pick html5 are we going to host the video?

zoek1 replied:
No, this is URL only. This could be hosted for example in skynet. I recently see some hackathon demos hosted in skynet.

octavioamu commented on app/dashboard/views.py:
```diff
@@ -5538,6 +5572,11 @@ def fulfill_bounty_v1(request):
+            demo_link = request.POST.get('videoDemoLink')
+            if demo_link:
+                project.extra['video_url'] = demo_link
+                project.extra['video_provider'] = demo_provider if demo_provider in ('loom', 'youtube', 'vimeo') else 'generic'
```
do we need the user to pick the provider? why not just do a regex to avoid users input errors and less fields on the form?

zoek1 replied:
Do you think is better to remove the provider select button? I think another option is to deactivate the button and display the provider based on the regex. Also in the current implementation regexs are used to determine the provider but the user can change it if he/she thinks the provider was wrong picked.

octavioamu replied:
yes based on regex should be better, the user shouldn't need to decide as the info is already on the link, if the link is not supported then let the user know. Sounds good?

danlipert replied:
Are we going to make this change soon so we can get this merged in? @zoek1

zoek1 replied:
done @danlipert @octavioamu

octavioamu reviewed (changes requested):
some questions Im not sure

danlipert reviewed (changes requested):
Once we remove the old provider selection UX then we are good to go 👍

danlipert commented:
I spent some time testing this today and noticed quite a few issues - I don't know what happened here but I was hoping to get this merged in today. I know a lot of people are waiting for this and it still has a bunch of unaddressed feedback.  Here's some of my test results:

- Setting the video url actually sets the work url (see app/dashboard/templates/addinterest.html L129)
- When editing a project, some fields that were previous optional are now required: category, built with, maybe others?

See also @chibie and @thelostone-mc 's comments regarding the `keyup` not working

danlipert reviewed (changes requested):
Found some bugs while testing, check my comments

octavioamu commented on app/dashboard/templates/bounty/fulfill.html:
```diff
@@ -93,6 +93,13 @@ <h4 class="text-center mb-4">{% trans "Submit Work" %}</h4>
+                      <label class="form__label" for="videoDemoLink">{% trans "Video demo Link" %}</label>
+                      <input name='videoDemoLink' id='videoDemoLink' class="form__input font-body" type="text"
+                        placeholder="https://youtu.be/DJartWzDn0E" value=""
+                        required
```
I think this should be required, not every bounty make sense to have a video

zoek1 replied:
Removed @octavioamu, this was required only for hackathon prizes (bounties)

zoek1 commented:
> I spent some time testing this today and noticed quite a few issues - I don't know what happened here but I was hoping to get this merged in today. I know a lot of people are waiting for this and it still has a bunch of unaddressed feedback. Here's some of my test results:
> 
> * Setting the video url actually sets the work url (see app/dashboard/templates/addinterest.html L129)
> * When editing a project, some fields that were previous optional are now required: category, built with, maybe others?
> 
> See also @chibie and @thelostone-mc 's comments regarding the `keyup` not working

@danlipert the id for video demo at `addinterest.html` Is now fixed.

Also the keyup functionality was dead code, no longer required since getting the required provider is now executed before sending the request. The keyup code is now removed.

zoek1 commented:
> When editing a project, some fields that were previous optional are now required: category, built with, maybe others?

@danlipert this was the interés behavior to force users to update the project metadata, i removed the required attribute in such elements to avoid this confusion in  the future.

danlipert commented:
@zoek1 see my fix here that was causing the problem with saving the video url https://github.com/gitcoinco/web/pull/7638/commits/a6f9d0e8be8e89163c17b33250c69e200ddcb53a
